[Refunds] Fix taxes handling

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: WOOMOB-395
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
This PR builds on top of the previous [PR](https://github.com/woocommerce/woocommerce-android/pull/14002) to fix the issue of tax refunding, as for now, we weren't handling it correctly.
This PR updates the logic of the refunding to handle splitting the taxes refund according to each item taxes.

**Important:** There is a discrepancy between the `taxes` array that each order line returns, and the `total_tax`, the `total_tax` is rounded, while the values in the array are raw. Initially, I wanted to avoid making any calculations on client side, and drop the `amount` argument from the API call, but I noticed an issue where the refunds could have less amount that the total order amount, so for now, we'll still send the `amount`, and its value will still use the `total_tax` for calculation (this matches iOS behavior too).
This issue will become problematic when we try to add support for partial refunds, I asked about it here p1747301522966739-slack-C07JJG089M0, and depending on the response, I'll document the next steps.

### Steps to reproduce
1. Enable tax calculation on your store.
2. Add multiple tax rates (make sure to assign different priorities so that they all get used for calculation).
3. Create an order.
4. Go to the app and refund the order.

### Testing information
- Confirm refunding works as expected.
- Check wp-admin and confirm taxes were correctly refunded (as shown in the screenshots below)

### The tests that have been performed
^

### Images/gif
| Before | After |
| --- | --- |
| <img width="972" alt="Screenshot 2025-05-15 at 14 50 23" src="https://github.com/user-attachments/assets/01b7e699-6cb0-453f-832c-8cf85e1b990c" /> | <img width="972" alt="Screenshot 2025-05-15 at 14 51 01" src="https://github.com/user-attachments/assets/50815edc-020e-428b-a50b-1ae4a0feef07" /> |

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Start updating models for unified refund calculation

hichamboushaba committed:
Remove more unused code

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Expose taxes in domain models

hichamboushaba committed:
Calculate tax IDs when refunding items

hichamboushaba committed:
Calculate tax IDs when refunding shipping and fees

hichamboushaba committed:
Add unit tests

hichamboushaba committed:
Pass the total amount calculated in the ViewModel to the API

The `taxes` array returned by the API includes raw taxes, not rounded, while the `total_tax` property is using rounded values, and this leads to some inconsistencies for calculating the amount when using them, so for now, we calculate the total amount using `total_tax` in the ViewModel.

hichamboushaba committed:
Fix detekt issues

hichamboushaba committed:
Add release note

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/refunds/RefundsUiModels.kt:
```diff
@@ -5,69 +5,87 @@ import com.woocommerce.android.model.Order
-) : Parcelable {
-    val availableRefundQuantity
-        get() = maxQuantity.toInt()
+sealed class RefundItem : Parcelable {
```
This refactoring is not really needed for this PR, but it will be useful for a future refactoring of the IssueRefundViewModel.

hichamboushaba committed:
Ensure product taxes total is based on the order total_tax

dangermattic commented:
<!--
  1 Error: This PR is tagged with <code>s...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>status: do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes reviewed (approved):
Nicely done @hichamboushaba the changes look good and work as described.
